packages/commonwealth/client/scripts/views/pages/notification_settings/index.tsx

```diff
@@ -99,7 +121,134 @@ const NotificationSettingsPage = () => {
+                    label="Receive Emails"
+                    disabled={true}
+                    checked={false}
+                    onChange={() => {
+                      app.user.notifications
+                        .enableImmediateEmails([])
+                        .then(() => {
+                          redraw();
+                        });
+                    }}
+                  />
+                  <CWToggle
+                    checked={false}
+                    onChange={() => {
+                      app.user.notifications
+                        .subscribe(NotificationCategories.ChainEvent, chain.id)
+                        .then(() => {
+                          redraw();
+                        });
+                    }}
+                  />
+                </div>
+              </div>
+            );
+          })}
+          {Object.entries(chainEventSubs).map(([chainName, subs]) => {
+            const chainInfo = app.config.chains.getById(chainName);
+            const hasSomeEmailSubs = subs.some((s) => s.immediateEmail);
+            const hasSomeInAppSubs = subs.some((s) => s.isActive);
+            return (
+              <div className="notification-row chain-events-subscriptions-padding">
+                <div className="notification-row-header">
+                  <div className="left-content-container">
+                    <div className="avatar-and-name">
+                      <CWCommunityAvatar size="medium" community={chainInfo} />
+                      <CWText type="h5" fontWeight="medium">
+                        {chainInfo?.name}
+                      </CWText>
+                    </div>
+                  </div>
+                  <CWCheckbox
+                    label="Receive Emails"
+                    checked={hasSomeEmailSubs}
+                    onChange={() => {
+                      hasSomeEmailSubs
+                        ? app.user.notifications
+                            .disableImmediateEmails(subs)
+                            .then(() => {
+                              redraw();
+                            })
+                        : app.user.notifications
+                            .enableImmediateEmails(subs)
+                            .then(() => {
+                              redraw();
+                            });
+                    }}
+                  />
+                  <CWToggle
+                    checked={subs.some((s) => s.isActive)}
+                    onChange={() => {
+                      hasSomeInAppSubs
+                        ? app.user.notifications
+                            .disableSubscriptions(subs)
+                            .then(() => {
+                              redraw();
+                            })
+                        : app.user.notifications
+                            .enableSubscriptions(subs)
+                            .then(() => {
+                              redraw();
```

Comment:
These redraws are because the original component never got refactored i.e. still has these redraws. I fixed the conflicts so that functionally the component should work but I have no idea what "reactifying" this component to remove the redraws actually entails. That is why I marked redraws as an outstanding issue.

Reply:
As you can see in the video, checkbox and toggles do not work because of the redraw implementation. Couple weeks ago Mark was refactoring this particular page to make it work (reactified it) so the checkoboxes, toggles and other stuff in `Discussion` section work well - you can see the [PR here](https://github.com/hicommonwealth/commonwealth/pull/3146).

You can either try to fix it in this PR or to push things further, we can merge it as it is, but please create a separate ticket that describes the issue and we can fix it as a follow up!

https://user-images.githubusercontent.com/14819225/230055229-7b192f25-1c1c-4b20-9474-feac02e3b72d.mp4

Reply:
The page doesn't even load on the React branch right now so I assumed it wasn't converted but thanks for pointing out the PR. It was very helpful to explicitly see Mark's changes in order to refactor my changes. Should be all good now, I reused the functions mark defined and all the toggles work well.